solr/core/src/java/org/apache/solr/core/CoreContainer.java

```diff
@@ -1018,8 +1040,52 @@ private void loadInternal() {
         "unloaded",
         SolrInfoBean.Category.CONTAINER.toString(),
         "cores");
+
     Path dataHome =
         cfg.getSolrDataHome() != null ? cfg.getSolrDataHome() : cfg.getCoreRootDirectory();
+
+    solrMetricsContext.observableLongGauge(
+        "solr_cores_filesystem_disk_space",
```

Comment:
I suggest calling this "solr_disk_space", (maybe "node" namespaced too).  
Practically speaking, I think this is the only disk space metric that matters.